jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementCharacteristic.kt

```diff
@@ -0,0 +1,142 @@
+package org.ehealthinnovation.jdrfandroidbleparser.bgm.characteristic.compoundcharacteristic
+
+import android.bluetooth.BluetoothGattCharacteristic
+import android.util.Log
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.characteristic.BaseCharacteristic
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.FormatType
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.GattCharacteristic
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.Units
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.Flags
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.SampleLocation
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.SensorStatusAnnunciation
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.encodedvalue.bgmmeasurement.Type
+import org.ehealthinnovation.jdrfandroidbleparser.bgm.utility.BluetoothDateTime
+import java.util.*
+
+class GlucoseMeasurementCharacteristic(characteristic: BluetoothGattCharacteristic?) : BaseCharacteristic(characteristic, GattCharacteristic.GLUCOSE_MEASUREMENT.assigned) {
+    var flags: EnumSet<Flags>? = null
+    var sequenceNumber: Int? = null
+    /**
+     * The year, month, day, hour, minute, second have value according to https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.date_time.xml
+     * These are the raw value extracted from the binary packet and not directly usable without proper conversion.
+     * These values are converted into a [Date] as [displayTime]
+     */
+    var timeOffset: Int? = null
+    var timeYear: Int? = null
+    var timeMonth: Int? = null
+    var timeDay: Int? = null
+    var timeHour: Int? = null
+    var timeMinute: Int? = null
+    var timeSecond: Int? = null
+    var displayTime: Date? = null
+
+    /**
+     * The concentration is the raw value from the binary packet. It has unit stored in [unit].
+     * A convenient function [getConcentration] to get the value in desired unit can be used to query this value.
+     */
+    var concentration: Float? = null
+    var unit: Units? = null
+
+    /**
+     * Contains a set of flag concerning the annunciation. Before interpreting the glucose concentration, make sure
+     * no flags indicating measurement error is present in this EnumSet.
+     */
+    var annunciationFlags: EnumSet<SensorStatusAnnunciation>? = null
+
+    /**
+     * The location and type of the blood sample.
+     */
+    var fluidType: Type? = null
+    var sampleLocation: SampleLocation? = null
+
+
+    /**
+     * This function is called when creating an characteristic to parse the binary array stream
+     */
+    override fun parse(c: BluetoothGattCharacteristic): Boolean {
+        var errorFreeParse = false
+        try {
+            //make sure c is not null
```

Comment:
c cannot be null here

Kotlin is null safe, you can't pass null into the method.